### PR TITLE
MINOR: Improve unit test coverage of LeaderAndIsr class

### PR DESCRIPTION
**Pull request by see-quick**
Signed-off-by: morsak <xorsak02@stud.fit.vutbr.cz>

This PR adds a few test cases for `LeaderAndIsr` class. Currently, on the UNIT level, we check `newRecoveringLeaderAndIsr` method. Therefore I have created mostly all unit tests for such class to improve coverage.

### CODE REVIEW

File: core/src/test/scala/kafka/api/LeaderAndIsrTest.scala [hunk 1]
```diff
@@ -31,4 +31,56 @@ final class LeaderAndIsrTest {
...
+    val leaderAndIsr = LeaderAndIsr(2, List(1, 2, 3))
+
+    assertEquals(2, leaderAndIsr.leaderOpt.get)
+  }
+
+  @Test
+  def testWithZkVersion() : Unit = {
+    val leaderAndIsr = LeaderAndIsr(3, List(1, 2, 3))
+
+    assertEquals(0, leaderAndIsr.zkVersion)
```
Comment (divijvaidya) [inline core/src/test/scala/kafka/api/LeaderAndIsrTest.scala]: `LeaderAndIsr` on trunk[ does not have a member called `zkVersion`](https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/api/LeaderAndIsr.scala#L22) and the compilation for this PR is failing due to that.

You might want to work with `trunk` branch in your local repository to get the latest code and verify the build succeeds using `/gradlew unittest`

Comment (see-quick) [reply core/src/test/scala/kafka/api/LeaderAndIsrTest.scala]: Yes, I have removed such a test case. It should be okay now.

Comment (see-quick) [reply core/src/test/scala/kafka/api/LeaderAndIsrTest.scala]: Thanks @divijvaidya

**Review by mimaison — APPROVED** [review]
LGTM

**Pull request merged**